### ISSUE TITLE
Dynamic select options in the SDC

### ISSUE THREAD

**Issue by pld** [opensrp/fhircore#593]
**Describe the issue to be researched**
How do we support having a multi-choice question where the values are populated and chosen from a list of some query against other resources?

Specifically, in G6PD, we could have a question “G6PD device” and the options rendered are a single select of any “DeviceDefinition” resources present in the local db. Does SDC support something like this now? If not, is there an existing extension we can use? (I think this nice to have category, but not sure, may become a req)

**Describe the goal of the research**
Goal is to decide on a single approach for a first implementation.

Artifacts to be produced are
- A description of the pros and cons of the different approaches for dynamic selection option loading
- Maybe PoCs of the different approaches, if needed

**Describe the methodology**
From @maimoonak:
> SDC supports resource population based on Expression, Observation, and StructureMap. http://build.fhir.org/ig/HL7/sdc/populate.html
@ekigamba has done some work on StructureMap population, and he can comment on current feature flexibility.
Where can more information be found? Who should the assignee approach to ask questions? How can a decision be made?

From @ekigamba:
> This is currently not supported with Expression-based population. This requires implementation of sdc-questionnaire-candidateExpression for choice selection in expression-based population

**Comment by pld** [comment]
Since we don't know if this is required yet, marking it blocked by that and removing from milestone

**Comment by mberg** [comment]
This is something we also would like for LMH.  One use case is populating a form with household members.

Jing shared this. He mentioned it has been implemented.

https://github.com/google/android-fhir/issues/613

https://github.com/google/android-fhir/pull/754

**Comment by pld** [comment]
Is this required for LMH?
